src/pin.rs

```diff
@@ -0,0 +1,89 @@
+use crate::geometry::*;
+use crate::text::*;
+
+#[allow(dead_code)]
+#[derive(Debug, Copy, Clone)]
+pub enum ElectricKind {
+    Input,
+    Output,
+    Bidirectional,
+    Tristate,
+    Passive,
+    Unspecified,
+    PowerInput,
+    PowerOutput,
+    OpenCollector,
+    OpenEmitter,
+    NotConnected,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Copy, Clone)]
+pub enum PinShape {
+    Line,
+    Inverted,
+    Clock,
+    InvertedClock,
+    InputLow,
+    ClockLow,
+    OutputLow,
+    FallingEdgeClock,
+    NonLogic,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Copy, Clone)]
+pub enum PinOrientation {
+    Up,
+    Down,
+    Right,
+    Left,
+}
+
+#[derive(Debug)]
+pub struct Pin {
+    pub pos: Point,
+    pub length: f64,
+    pub net: String,
+    pub number: u32,
```

Comment:
`number` may be a `String` for BGA packages

Reply:
Whoa. Thank you. That was a little bit unexpected. Made it a ``String`` too.